app/uk/gov/hmrc/fileupload/controllers/RedirectionFeature.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.fileupload.controllers
+
+
+import java.net.{MalformedURLException, URL}
+
+import play.api.http.HttpEntity
+import play.api.http.Status.{BAD_REQUEST, MOVED_PERMANENTLY}
+import play.api.mvc._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success, Try}
+
+
+class RedirectionFeature(allowedHosts: Seq[String]) {
+  import EnvelopeChecker.logAndReturn
+
+  case class ValidatedUrl(url: String)
+  case class RedirectUrlsO(succ: Option[ValidatedUrl], fail: Option[ValidatedUrl])
+
+  def redirect(successUrlO: Option[String], failureUrlO: Option[String])
```

Comment:
if the type is not given explicitly then without Intelij it is hard to get the type, f.ex. at github reviews.
This PR code might be better without need of such naming but no time for that.